src/app/src/main/kotlin/org/vaccineimpact/api/app/logic/ResponsibilitiesLogic.kt

```diff
@@ -36,4 +38,59 @@ class RepositoriesResponsibilitiesLogic(
 
         scenarioRepository.checkScenarioDescriptionExists(path.scenarioId)
     }
-}
\ No newline at end of file
+
+    override fun getTouchstoneResponsibilitiesData(touchstoneVersionId: String): List<ResponsibilitiesRow>
+    {
+        val comments = responsibilitiesRepository.getResponsibilitiesWithCommentsForTouchstone(touchstoneVersionId)
+        val sets = responsibilitiesRepository.getResponsibilitiesForTouchstone(touchstoneVersionId)
+        return sets.flatMap { set ->
+            val responsibilitySetComment = comments.find { it.modellingGroupId == set.modellingGroupId}?.comment
+            set.responsibilities.map { responsibility ->
+                val currentEstimateSet = responsibility.currentEstimateSet
+                val responsibilityComment = comments.find { it.modellingGroupId == set.modellingGroupId }?.responsibilities?.find { it.scenarioId == responsibility.scenario.id }?.comment
+                val responsibilityInfo = burdenEstimateRepository.getResponsibilityInfo(set.modellingGroupId, touchstoneVersionId, responsibility.scenario.id)
+                val expectationMapping = expectationsRepository.getExpectationsForResponsibility(responsibilityInfo.id)
+                val validatedRowMap = if (currentEstimateSet != null) burdenEstimateRepository.validateEstimates(currentEstimateSet, expectationMapping.expectation.expectedRowLookup()) else null
+                ResponsibilitiesRow(
```

Comment:
I think a couple of comments would be useful here, for the less obvious data derivations here, particularly `current_estimate_set_missing_estimates` - I think this returns the count of all the years and ages where the validatedRowMap returns false (invalid)?